rcgen.js

```diff
@@ -0,0 +1,88 @@
+// @ts-check
+
+const {gitFiles} = require('./rcgen/git/files');
+const {ignoreFilesWithGit} = require('./rcgen/git/utils');
+const {prettierFiles} = require('./rcgen/prettier/files');
+const {prettierPatchers} = require('./rcgen/prettier/patchers');
+const {
+  configurePrettier,
+  ignoreFilesWithPrettier
+} = require('./rcgen/prettier/utils');
+const {vscodeFiles} = require('./rcgen/vscode/files');
+const {vscodePatchers} = require('./rcgen/vscode/patchers');
+const {
+  excludeFilesFromSearchInVscode,
+  excludeFilesInVscode,
+  recommendVscodeExtensions
+} = require('./rcgen/vscode/utils');
+
+/**
+ * @type {import('@rcgen/core').Manifest}
+ */
+exports.default = {
+  files: [...gitFiles, ...prettierFiles, ...vscodeFiles],
+  patchers: [
+    ...prettierPatchers,
+    ...vscodePatchers,
+    ignoreFilesWithGit(
+      '.cache',
+      'coverage',
+      'lerna-debug.log',
+      'lib',
+      'node_modules',
+      'npm-debug.log',
+      'package-lock.json',
```

Comment:
why would I .gitignore a lockfile?

Reply:
This file doesn't make sense in a Yarn Repo.